### PR TITLE
Support for gcp-login

### PR DESCRIPTION
**Pull request by idodod**
This implements support for using gcp-login with a gcloud service account json config.
Based on steps described [here](https://cloud.google.com/artifact-registry/docs/docker/authentication#token).

Similar to ecr-login, this extends the notion of storing credentials under std/registry/<host>, by first looking for a cred_helper file, and in this PR, adding another option - `gcp-login`, which can be used for either  Google Artifact Registry or Google Container Registry.

If the value is `gcp-login`, we will read from std/registry/\<host\>/GCP_KEY secret, and pass those to the gcp-login credential library. Note that it is not required to have `gcloud` or `docker-credential-gcr` installed anywhere on your PATH.

~Note: waiting for https://github.com/earthly/earthly/pull/2668 to be merged first.~ - Done

### CODE REVIEW

File: util/llbutil/authprovider/cloudauth/gcp_login.go
Comment (alexcb) [inline util/llbutil/authprovider/cloudauth/gcp_login.go:31]: missing err handling?

File: tests/registry-command/test-gcp.sh
Comment (alexcb) [inline tests/registry-command/test-gcp.sh:23]: awesome to see it follows the same test-flow :tada:

File: util/llbutil/authprovider/cloudauth/gcp_login.go
Comment (idodod) [reply util/llbutil/authprovider/cloudauth/gcp_login.go:31]: good catch, looks like I removed it by mistake.
That makes me wonder though - should I add integration tests to for the err path (secret is missing, json in the secret is invalid, etc)?

File: util/llbutil/authprovider/cloudauth/ecr_login.go
Comment (idodod) [inline util/llbutil/authprovider/cloudauth/ecr_login.go:14]: Using a constant but I can undo this change (made it to match with `GCPCredHelper`)

File: tests/registry-command/test-ecr.sh
Comment (alexcb) [inline tests/registry-command/test-ecr.sh]: For some reason I thought this trap required bash, but if that's not the case, then there's no need for `env` either. (I only used `env` since `bash` as a matter of habit since it doesn't always get installed in the same location)
```suggestion
#!/bin/sh
```

Comment (idodod) [reply tests/registry-command/test-ecr.sh]: sorry, that was just a mistake after I rebased, I reverted this change for now.

File: util/llbutil/authprovider/cloudauth/ecr_login.go
Comment (alexcb) [reply util/llbutil/authprovider/cloudauth/ecr_login.go:14]: nice change!

**Review by alexcb — APPROVED** [review]

**Review by nelsam — APPROVED** [review]
It all seems reasonable to me. I wanted to wait until Alex's PR was merged to avoid reviewing his changes amidst yours, but yeah, lgtm.

**Pull request merged**
